Change arg of getSubGroups to briefRepresentation

transducer opened the pull request:
Parameter name briefRepresentation should be briefRepresentation, not full. This way callers will by default get the full representation unless true is passed as value for briefRepresentation.

Fixes #25096

Mistake introduced in 23.0.0 in commit https://github.com/keycloak/keycloak/commit/69497382d8cc6c622e815184d9bc27cb0dc6a400#diff-d233e82db465ac2b1ad22b124ab0bae3714f02d58c108c29afbbb935653d3ceaR162.

I could not find a place to add an integration test for testing the endpoint with an external caller. If such a test exists please point me to it and I will add it.

transducer commented:
@mhajas yes please let me know where I can add it. I actually don't think the current commit fixes https://github.com/keycloak/keycloak/issues/25053, since GET {keycloak server}/realms/{realm}/groups/{group_id}/children seems to be a new endpoint, and the issue states the older GET {keycloak server}/realms/{realm}/groups/ endpoint fails.

mhajas commented:
@transducer Yes you are right. this does not fix #25053. Please remove it from the commit message and PR description so it is not linked to this PR. Also, Keycloak now requires [Developer's Certificate of Origin](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#developers-certificate-of-origin) otherwise the CI won't let this PR in.

Regarding the test, the correct file seems to be: https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/group/GroupTest.java

We are interested in doing a call like this: https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/group/GroupTest.java#L427 specifically the method `getSubGroups` with `briefRepresentation` set to `true` and `false` and then check the level of details within the returned list of `GroupRepresentation`s. Add a new test case for that please because some of the test cases already there are huge.

mposolda commented:
@mhajas @transducer 2 questions:

- Do we want this change documented in the upgrading guide? Something like `In Keycloak 23.0.0 release, we accidentally changed the meaning of parameter in briefRepresentation. This is fixed now, so the briefRepresentation parameter has same meaning like before Keycloak 23 release` or something like this?

- Do we want this backported to Keycloak 23 as well, so it is available in next Keycloak 23.0.X release? I suppose that yes. In this case, the appropriate place for documenting this change might be file https://github.com/keycloak/keycloak/blob/main/docs/documentation/upgrading/topics/keycloak/changes-23_0_2.adoc

transducer commented:
@mposolda The endpoint GET /groups/<group_id>/children was introduced in 23.0.0, this is also mentioned in the [previous release notes](https://github.com/keycloak/keycloak/commit/69497382d8cc6c622e815184d9bc27cb0dc6a400#diff-430c07d2628f0b5fc42d0999559c52d55c1972e64d4c814415c4248e41a4e23cR134). So I can add it to the notes for this PR.

As mentioned, the regression that the meaning was changed for other endpoints too (mentioned in https://github.com/keycloak/keycloak/issues/25053), will not be fixed by this PR.

I will add a unit test and a release note later today.

mposolda commented:
@transducer Thanks!

transducer commented:
@mposolda I've added a unit test and documentation.

I unfortunately made a mistake where I rebased a previous commit which caused extra reviewers to be automatically added. I removed this commit but the added labels stay.

I don't know if the test works, because I don't have a working Java IDE and locally I get a build error when running `mvn clean install`, and the CI does not run.

ahus1 commented:
@transducer - for first-time contributors the CI doesn't run automatically but has to be approved by one of the maintainers. This is the case for every new push to the PR until the you've had your first PR merged. This is a precaution built into standard GitHub. 

I've just now approved the run, and CI is starting.

transducer commented:
@mposolda I think test should work now. Since I could not run the project locally I am using the CI to check. Could you run it once more?

transducer commented:
@mposolda sorry it's a bit inconvenient using the CI to check, the order in the check was wrong, now using hamcrest.Matchers it should be fine. Could it be started once again?

transducer commented:
@ahus1 could you approve one more run of this PR? I think the test is okay now (cannot test locally).

mposolda commented:
@transducer Sorry, one small thing: Keycloak 23.0.2 has already code-freeze for past days, so this PR won't be possible to catch in that release. Is it possible to put it to `changes-23_0_3.adoc` instead? Also might be needed to include this file from `changes.adoc` as it is not created yet.

transducer committed:
Change arg of getSubGroups to briefRepresentation

   Parameter name briefRepresentation should mean briefRepresentation,
   not full. This way callers will by default get the full
   representation, unless true is passed as value for
   briefRepresentation.

   Fixes #25096

Signed-off-by: Erwin Rooijakkers <erwin@rooijakkers.software>

transducer commented:
@mposolda done

transducer commented:
@mposolda Now Store Models Test is failing ([logs](https://github.com/keycloak/keycloak/actions/runs/7168641214/job/19587997734?pr=25264#logs)). I searched for "Errors: 1" (Failures are all 0) and found: `7:56:50,408 ERROR [org.keycloak.models.sessions.infinispan.initializer.InfinispanCacheInitializer] (pool-50-thread-3) Interruped exception when computed future. Errors: 1` I don't think this is related to my change?

mposolda reviewed (approved):
@transducer Yes, I also think it is not related to your change. Will try to re-run. Hope it helps...
